### ISSUE TITLE
Unicode 32 fonts (Material Design) are not rendering properly

### ISSUE THREAD

**Issue by mhamid3d** [spyder-ide/qtawesome#217]
All the fonts provided in qtawesome are using Unicode 16 BMP fonts except for Material Design which is using Unicode 32. For some reason the unicodes are getting truncated in some QT versions and all the icons show up as the garbage "?" icon.

This is happening in PySide2 v5.15.6 if you want to try to reproduce the bug.

I made a bug with QT but they are claiming that this issue is with how we are handling the unicode characters in python, causing them to be truncated. At this point I am making this issue here to see if there are any ideas of how we can adjust the code to find a solution for this.

**Comment by dalthviz** [comment]
Hi @mhamid3d thank you for the feedback! I was able to reproduce this behavior on Windows with PySide2 v5.51.6. Checked also on Windows with PyQt5 v5.15.6 but with that setup I'm able to see the icons correctly.

Checked with QtAwesome 0.7.3, 1.0.3, 1.1.1 and current dev (1.3.0.dev0) and the result is the same except for v0.7.3 (so maybe the problem started when we changed MDI from 4.x to 5.x?) 🤔 

Previews:

<details>

### PySide2 v5.15.6 

#### QtAwesome v0.7.3

![imagen](https://user-images.githubusercontent.com/16781833/197828167-ef68999b-5bc7-4c1a-b466-9a74fcc1cd66.png)

#### QtAwesome v1.0.3

![imagen](https://user-images.githubusercontent.com/16781833/197827680-41320a6c-39a1-457b-adc8-e1a33ae6a9ac.png)

#### Current QtAwesome dev

![imagen](https://user-images.githubusercontent.com/16781833/197829065-45fc5951-a61e-4300-9332-ede14311b62d.png)


### PyQt5 5.15.6

#### QtAwesome v0.7.3

![imagen](https://user-images.githubusercontent.com/16781833/197828578-53a9ab7f-ee9f-4ed5-96a1-4d65e87cd39e.png)

#### QtAwesome v1.0.3

![imagen](https://user-images.githubusercontent.com/16781833/197828757-3deae72c-e74a-4687-8550-c7dafd09b007.png)

#### Current QtAwesome dev

![imagen](https://user-images.githubusercontent.com/16781833/197829218-eaf1591a-4bc7-4a75-8a90-a0ecb433137e.png)


</details>


Not totally sure what could be happening :/ but maybe this means we need to check how the material design font is generated at source? We take the font from https://raw.githubusercontent.com/Templarian/MaterialDesign-Webfont/master/fonts/materialdesignicons-webfont.ttf and seems like is generated using https://github.com/fontello/svg2ttf

Maybe @ccordoba12 or @kumattau have suggestions to further debug or fix this?

**Comment by ccordoba12** [comment]
This seems to me a bug in PySide2 given that QtAwesome works just fine in PyQt5.

**Comment by mhamid3d** [comment]
Hey @dalthviz @ccordoba12, I was able to confirm this is an issue on how PySide2 was built, a flag is omitted on build that would've allowed support for 32bit unicode, instead now they are being truncated to 16-bit. Since MDI uses 32bit, it is the only font with issues (I tried to ask them for a 16-bit alternative but they are fully committed to 32-bit at this point).

I asked on the QT bug thread and they confirmed there is nothing we can do on the python end to work around this bug, as such I will close this. Thanks a lot for checking that!

https://bugreports.qt.io/browse/PYSIDE-1835
